README.adoc

```diff
@@ -145,6 +145,15 @@ public interface TestRetryTaskExtension {
      */
     Property<Integer> getMaxFailures();
 
+    /**
+     * Whether test classes should be retried in entirety, or if individual tests methods can be retried.
+     * <p>
+     * This setting defaults to {@code false}, meaning that only the test methods that actually fail will be retried.
+     *
+     * @return whether individual test methods or entire test classes should be retried
+     */
+    Property<Boolean> getRetryEntireTestClass();
```

Comment:
Are we sure we want to do it this “coarsely”? Should we instead allow classes to opt-in via an extension? I think the current proposed is OK, but wanted to put the question on the record.

Reply:
From my point of view, when you know that this or that class/set of classes requires retries, this is indeed something that could be defined as an annotation on a class but till now there are no annotations/anything that is exposed directly in tests. 

Another approach to set a filter in gradle config would also be fine. If we relaunch everything that may lead to unnecessary overhead.

Reply:
I agree that annotating a class to be retried in entirety (or alternatively registering the class with the `retry` extension) would be a better, more fine-grained solution. We could take a similar approach to flag particular tests as needing a retry, rather than doing this in a coarse-grained manner.

I'm definitely open to this idea, but I don't think I'll have capacity to do the implementation any time soon.